pkg/controllers/proxy/network_services_controller.go

```diff
@@ -910,6 +905,11 @@ func (nsc *NetworkServicesController) syncIpvsServices(serviceInfoMap serviceInf
 		var clusterServiceId = generateIpPortId(svc.clusterIP.String(), svc.protocol, strconv.Itoa(svc.port))
 		activeServiceEndpointMap[clusterServiceId] = make([]string, 0)
 
+		if svc.local && !hasActiveEndpoints(svc, endpoints) {
+			glog.V(1).Infof("Skipping NodePort/LoadBalancer service %s/%s as it does not have active endpoints\n", svc.namespace, svc.name)
+			continue
```

Comment:
I think that the problem is that if you `continue` here, then you end up with an IPVS service without any endpoints in it as endpoints are added below in the other section you modified.

So you need to make it to the section below, but you don't want to run anything imbetween. Probably the easiest way would be to abstract the nodeport and externalIP stuff into their own methods, then gate them by a variation of this if statement.